composer/helpers/v2/src/UpdateChecker.php

```diff
@@ -86,7 +77,7 @@ public static function getLatestResolvableVersion(array $args): ?string
 
         $install->run();
 
-        $installedPackages = $installationManager->getInstalledPackages();
+        $installedPackages = $composer->getLocker()->getLockedRepository(true)->getPackages();
```

Comment:
This argument is "withDevReqs" to return all packages including dev requirements.